src/processor/tree/tree_wrapper.rs

```diff
@@ -79,35 +96,68 @@ impl TreeWrapper {
             hex::encode(root_hash)
         );
 
-        assert_eq!(root_hash.as_bytes(), block.new_state_root);
-        tracing::debug!("Successfully processed block {}", block.l2_block_number);
+        let root_hash_bytes = root_hash.as_bytes();
```

Comment:
We should extract this part into a separate function. E.g. `fn ascertain_state_consistency()`.

Reply:
well, why? It isn't like the rollback is going to be called from multiple places...

Reply:
Of course not, I was thinking that it's more for organization and maintainability reasons.

Reply:
well, I don't think a separate function would make the code more maintainable... It isn't just that it wouldn't be called from elsewhere, but that it _couldn't_ - it depends on the correct state of `index2key` and `key2value`, which are updated in multiple places...

Reply:
Fair point. Let's leave as is then